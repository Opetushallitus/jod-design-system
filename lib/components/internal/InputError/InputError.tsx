interface InputErrorProps {
  /** The error message to display below the input field */
  errorMessage?: string;
  /** The id for the error message element */
  id: string;
  /** Test id for querying in tests */
  dataTestId?: string;
}

export const InputError = ({ errorMessage, id, dataTestId }: InputErrorProps) => {
  return errorMessage !== undefined ? (
    <div
      id={id}
      className="ds:mt-2 ds:block ds:text-form-error ds:text-alert-text-2 ds:font-arial"
      role="alert"
      data-testid={dataTestId}
    >
      {errorMessage}
    </div>
  ) : null;
};
