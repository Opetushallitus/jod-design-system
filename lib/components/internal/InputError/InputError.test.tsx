import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { InputError } from './InputError';

describe('InputError', () => {
  const errorMessage = 'This is an error message';
  const id = 'error-message';
  const dataTestId = 'input-error';

  it('renders error message correctly', () => {
    const { container } = render(<InputError errorMessage={errorMessage} id={id} dataTestId={dataTestId} />);
    expect(screen.getByTestId(dataTestId)).toBeInTheDocument();
    const errorElement = screen.getByText(errorMessage);
    expect(errorElement).toBeInTheDocument();
    expect(errorElement).toHaveAttribute('id', id);
    expect(container).toMatchSnapshot();
  });
  it('does not render when errorMessage is undefined', () => {
    const { container } = render(<InputError id={id} />);
    expect(container).toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
