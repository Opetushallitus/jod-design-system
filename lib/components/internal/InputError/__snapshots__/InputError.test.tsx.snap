// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputError > does not render when errorMessage is undefined 1`] = `<div />`;

exports[`InputError > renders error message correctly 1`] = `
<div>
  <div
    class="ds:mt-2 ds:block ds:text-form-error ds:text-alert-text-2 ds:font-arial"
    data-testid="input-error"
    id="error-message"
    role="alert"
  >
    This is an error message
  </div>
</div>
`;
